azurelinuxagent/common/protocol/wire.py

```diff
@@ -790,35 +781,35 @@ def _update_from_goal_state(self, refresh_type):
                 new_goal_state = GoalState(new_goal_state_xml)
 
                 def update_host_plugin():
-                    if self.host_plugin is not None:
-                        self.host_plugin.container_id = new_goal_state.container_id
-                        self.host_plugin.role_config_name = new_goal_state.role_config_name
+                    if self._host_plugin is not None:
+                        self._host_plugin.update_container_id(new_goal_state.container_id)
+                        self._host_plugin.update_role_config_name(new_goal_state.role_config_name)
 
                 if refresh_type == WireClient._UpdateType.HostPlugin:
                     update_host_plugin()
                     return
 
                 def incarnation_changed():
                     last_incarnation = None
-                    if os.path.isfile(incarnation_file):
-                        last_incarnation = fileutil.read_file(incarnation_file)
+                    if self._goal_state is not None:
```

Comment:
Previously we checked the last_incarnation by checking the file values, that made different WireClient objects out of sync. Checking the incarnation value of the current object to ensure that all objects in all threads are always in sync with the latest goal state